src/Compilers/CSharp/Portable/Symbols/Source/SourceEventAccessorSymbol.cs

```diff
@@ -122,9 +121,6 @@ protected sealed override void MethodChecks(BindingDiagnosticBag diagnostics)
 
                         _lazyReturnType = TypeWithAnnotations.Create(eventTokenType);
                         this.SetReturnsVoid(returnsVoid: false);
-
-                        var parameter = new SynthesizedAccessorValueParameterSymbol(this, _event.TypeWithAnnotations, 0);
-                        _lazyParameters = ImmutableArray.Create<ParameterSymbol>(parameter);
```

Comment:
Should we assert here that the types remain as expected? For example here:

```cs
Debug.Assert(_parameters is [var p] && p.TypeWithAnnotations == _event.TypeWithAnnotation);
```

Reply:
> Should we assert here and below that the types remain as expected? For example here

While we could do that. There will be a downside. That will mean that in DEBUG we will be synchronizing two processes that are going to be fully asynchronous in RELEASE, which might lead to a noticeable difference in behavior between the flavors. So, I am reluctant to add an assert like that.

Reply:
However, If you feel strongly about having the assert, I will add it.

Reply:
Thanks for the explanation - now I think it's better without an assert here then.